src/System.Management.Automation/engine/DataStoreAdapterProvider.cs

```diff
@@ -50,35 +50,37 @@ internal string FullName
         {
             get
             {
-                string GetFullName(string name, string psSnapInName, string moduleName)
-                {
-                    string result = name;
-                    if (!string.IsNullOrEmpty(psSnapInName))
-                    {
-                        result =
-                            string.Format(
-                                System.Globalization.CultureInfo.InvariantCulture,
-                                "{0}\\{1}",
-                                psSnapInName,
-                                name);
-                    }
+                return _fullName ?? (_fullName = GetFullName());
```

Comment:
The name is still cached. But does it make sens to caching this name at all?

Reply:
This comes from #8831 where we add the optimization. It is amazing to calculate the full name for every provider operation. I'd do not revert the optimization.

Reply:
So it's not a problem to use the cached name in other places but only those 2 spots you changed in this PR?

Reply:
It is important to do not cache the provider full name in the submodule until we set root module name.

Reply:
Then it would be very helpful to add comments to the place where you use `GetFullName()` instead of the cached fullname, otherwise, another refactoring in future might change it back to use the cached value.

Reply:
The comment is already there "Use GetFullName() to avoid caching full name at init time."

Reply:
Ah, yes, I missed that. Can you please also add the information about why to avoid the cached name?